pkg/controller/klusterletaddon/globalproxy_controller.go

```diff
@@ -241,10 +296,49 @@ func getGlobalProxyInInstallConfig(installConfig []byte) (agentv1.ProxyConfig, e
 	if err != nil {
 		return proxyConfig, err
 	}
-	proxyConfig.NoProxy, _, err = unstructured.NestedString(proxyConfigRaw, "proxy", "noProxy")
+
+	noProxy, _, err := unstructured.NestedString(proxyConfigRaw, "proxy", "noProxy")
+	if err != nil {
+		return proxyConfig, err
+	}
+
+	if proxyConfig.HTTPProxy == "" && proxyConfig.HTTPSProxy == "" && noProxy == "" {
+		return proxyConfig, nil
+	}
+
+	if noProxy == "*" {
+		proxyConfig.NoProxy = noProxy
+		return proxyConfig, nil
+	}
+
+	//noProxyList := []string{".cluster.local", ".svc", "localhost", "127.0.0.1"}
+	noProxyList := sets.NewString(".cluster.local", ".svc", "localhost", "127.0.0.1")
+	noProxyList.Insert(noProxy)
+
+	clusterNetworkCIDRs, err := getClusterNetworkCIDRs(proxyConfigRaw)
+	if err != nil {
```

Comment:
what if clusterNetworkCIDRs is not found or user does not set here?

Reply:
clusterNetworkCIDRs is required when create the cluster. 
there will meet error if clusterNetworkCIDRs is not found. will not append the proxy config to the status of klusterletaddonconfig, and output error info in the status condition.